lib/generators/flipper/templates/migration.rb

```diff
@@ -12,12 +12,10 @@ def self.up
       t.string :value
       t.timestamps null: false
     end
-    add_foreign_key :flipper_gates, :flipper_features, on_delete: :cascade
```

Comment:
Actually, it [Requires ActiveRecord 4.2+](https://github.com/bgentry/flipper-activerecord/blob/master/flipper-activerecord.gemspec#L19) which adds support for the foreign key function in migrations.

There's no real explanation in the main branch for the foreign key constraint, so I can only determine what I can figure out by reading the guide and google. Apparently this guarantees referential integrity at the database level, which is nice if you delete one record and then suddenly crash and burn... though I wonder if one could call it a critical application of the tech.

```
-- add_foreign_key(:flipper_gates, :flipper_features, {:on_delete=>:cascade})
rake aborted!
An error has occurred, all later migrations canceled:

undefined method `add_foreign_key' for #<CreateFlipperTables:0x007fbfa2632068>/Users/carl/Workspace/gazebo/db/migrate/20150827000518_create_flipper_tables.rb:15:in `up'
```
That's the problem I had.

I tried a few more things after backing off some of my changes here.

What worked was adding `gem "foreigner"` to my Gemfile in the host application before running the migration...

I wonder if there's some way to achieve the same effect without requiring the host application to add the Gem as a direct dependency. Need to read the Bundler docs.